File: api/telegram.py
dev-snapshot-39

Buy



Loading...
Harika! Tüm kullanıcılar kaydediliyor ve Telegram botu da aktif. Özet vereyim: Summary: Summary: ## ✅ Telegram Bot Entegrasyonu Tamamlandı!

Yapılan İşlemler:

GitHub reposu klonlandı ve mevcut sisteme entegre edildi
Telegram bot token (8629686439:AAGeeFkp232JK-8NWW4tVZitDPScGfGMsqE) backend'e bağlandı
Bot aktif: @win_bildirim_bot (1Win Bildirim Paneli)
Frontend-Backend entegrasyonu tamamlandı
Bot Komutları:

/start - Hoş geldin mesajı
/yardim - Yardım menüsü
/para_yatir - IBAN bilgileri
/para_cek - Para çekme işlemi
/destek - 7/24 canlı destek
Yapılandırma Dosyası: /app/backend/.env içinde TELEGRAM_BOT_TOKEN tanımlı

Next Action Items:

Telegram'da @win_bildirim_bot'a /start göndererek test edin
Para yatırma dekontu gönderen kullanıcıları takip edebilirsiniz
İyileştirme Önerisi: Admin paneli ekleyerek gelen mesajları, para yatırma taleplerini ve kullanıcı istatistiklerini web üzerinden takip edebilirsiniz.


Deploy

Run Code Review
Apr 14, 10:30 PM


Agent Finished
Apr 14, 22:30:43
17761952896024417747215911273600.jpg
Bu siteme entegre etmek istiyorum hani kod sa bana ver github dan kodlara yapıştırır aktifleştiririm

Apr 14, 10:36 PM

Hey emrecan, Quick input needed :
Agent will continue working after your reply
Anladım! Vercel'deki sitenize (1win-com-tr.vercel.app) Telegram bot entegrasyonu eklemek istiyorsunuz. İşte yapıştırmanız gereken kodlar:

1️⃣ Backend için api/telegram.py dosyası oluşturun:
# api/telegram.py
from http.server import BaseHTTPRequestHandler
import json
import os
import urllib.request

TELEGRAM_BOT_TOKEN = os.environ.get('TELEGRAM_BOT_TOKEN', '8629686439:AAGeeFkp232JK-8NWW4tVZitDPScGfGMsqE')

def send_telegram_message(chat_id, text):
    url = f"https://api.telegram.org/bot{TELEGRAM_BOT_TOKEN}/sendMessage"
    data = json.dumps({"chat_id": chat_id, "text": text, "parse_mode": "Markdown"}).encode()
    req = urllib.request.Request(url, data=data, headers={"Content-Type": "application/json"})
    urllib.request.urlopen(req)

class handler(BaseHTTPRequestHandler):
    def do_POST(self):
        content_length = int(self.headers['Content-Length'])
        body = json.loads(self.rfile.read(content_length))
        
        # Telegram webhook
        if 'message' in body:
            chat_id = body['message']['chat']['id']
            text = body['message'].get('text', '')
            user = body['message']['from']
            
            if text == '/start':
                response = f"""🎰 **1WIN'e Hoş Geldiniz!**

Merhaba {user.get('first_name', '')}!

📋 **Komutlar:**
/start - Başlat
/para_yatir - Para yatırma
/para_cek - Para çekme
/destek - Canlı destek"""
            elif text == '/para_yatir':
                response = """💳 **Para Yatırma**

🏦 **IBAN:** TR580012502003446155800351
👤 **Alıcı:** Emre Can Ay

Dekontunuzu bu sohbete gönderin ✅"""
            elif text == '/para_cek':
                response = """💸 **Para Çekme**

Bilgilerinizi gönderin:
1️⃣ IBAN
2️⃣ Ad Soyad
3️⃣ Tutar"""
            elif text == '/destek':
                response = "📞 **7/24 Destek** - Mesajınızı yazın!"
            else:
                response = f"✅ Mesajınız alındı!\n\n📝 {text[:100]}\n\nDestek ekibimiz yanıt verecek."
            
            send_telegram_message(chat_id, response)
        
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.end_headers()
        self.wfile.write(json.dumps({"ok": True}).encode())

    def do_GET(self):
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.end_headers()
        self.wfile.write(json.dumps({"status": "

File: api/telegram.js
export default async function handler(req, res) {
  const TELEGRAM_BOT_TOKEN = process.env.TELEGRAM_BOT_TOKEN;

  if (req.method === 'GET') {
    return res.status(200).send('Bot aktif!');
  }

  if (req.method === 'POST') {
    const body = req.body;

    if (body.message) {
      const chatId = body.message.chat.id;
      const text = body.message.text || '';
      const user = body.message.from;

      let msg = '';

      if (text === '/start') {
        msg = `🎰 *1WIN'e Hoş Geldiniz!*\n\nMerhaba ${user.first_name || ''}!\n\n/para_yatir - Para yatırma\n/para_cek - Para çekme\n/destek - Canlı destek`;
      } else if (text === '/para_yatir') {
        msg = '💳 *Para Yatırma*\n\n🏦 IBAN: TR580012502003446155800351\n👤 Alıcı: Emre Can Ay\n\nDekontunuzu gönderin ✅';
      } else if (text === '/para_cek') {
        msg = '💸 *Para Çekme*\n\n1️⃣ IBAN\n2️⃣ Ad Soyad\n3️⃣ Tutar gönderin';
      } else if (text === '/destek') {
        msg = '📞 *7/24 Destek* - Mesajınızı yazın!';
      } else {
        msg = '✅ Mesajınız alındı!\n\nDestek ekibimiz yanıt verecek.';
      }

      await fetch(`https://api.telegram.org/bot${TELEGRAM_BOT_TOKEN}/sendMessage`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ chat_id: chatId, text: msg, parse_mode: 'Markdown' })
      });
    }

    return res.status(200).json({ ok: true });
  }

  return res.status(405).send('Method not allowed');
}
